src/navx_node.cpp

```diff
@@ -67,7 +77,62 @@ int main(int argc, char** argv) {
   ros::spin();
 }
 
+void calculate_covariance(boost::array<double, 9> &orientation_mat, boost::array<double, 9> &ang_vel_mat, boost::array<double, 9> &accel_mat) {
+  int count = std::min(seq-1, covar_samples);
+  if (count < 2) {
+    return;
+  }
```

Comment:
I would have it return bool, whether it passed or not, and you can make the `2` be a tuning factor; 
That way you can choose not to publish the Imu msg until you have a covariance for it (should, ergo, take less than a sec, but no unclear data).